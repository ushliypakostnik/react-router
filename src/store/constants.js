export const FETCH_URL = process.env.REACT_APP_API_URL;

export const THEME = {
  DARK: "dark",
  LIGHT: "light"
};

export const COOKIES = {
  THEME: "theme"
};
