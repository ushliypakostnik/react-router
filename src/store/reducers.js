const reducer = (state = { }, action) => {
  switch (action.type) {
    case "REQUEST_ALBUMS":
      return Object.assign({}, state, {
        isFetching: true
      });
    case "RECEIVE_ALBUMS":
      return Object.assign({}, state, {
        isFetching: false,
        albums: action.albums
      });
    case "REQUEST_ALBUMS_FAILED":
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });
    case "REQUEST_DATA":
      return Object.assign({}, state, {
        isFetching: true
      });
    case "RECEIVE_DATA":
      return Object.assign({}, state, {
        isFetching: false,
        data: action.data
      });
    case "REQUEST_DATA_FAILED":
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });
    case "PAGE_TO_ACTIVE":
      return Object.assign({}, state, {
        activePage: action.activePage
      });
    case "TOGGLE_THEME":
      return Object.assign({}, state, {
        theme: action.theme
      });
    default:
      return state;
  }
}

export default reducer;
