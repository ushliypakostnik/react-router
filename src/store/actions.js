import { FETCH_URL } from './constants';

// Actions Types

export const PAGE_TO_ACTIVE = 'PAGE_TO_ACTIVE';
export const TOGGLE_THEME = 'TOGGLE_THEME';
export const RESIZE = 'RESIZE';

export const REQUEST_ALBUMS = 'REQUEST_ALBUMS';
export const RECEIVE_ALBUMS = 'RECEIVE_ALBUMS';
export const REQUEST_ALBUMS_FAILED = 'REQUEST_ALBUMS_FAILED';

export const REQUEST_DATA = 'REQUEST_DATA';
export const RECEIVE_DATA = 'RECEIVE_DATA';
export const REQUEST_DATA_FAILED = 'REQUEST_DATA_FAILED';

// Action Creators

export const requestAlbums = () => ({
  type: REQUEST_ALBUMS
});

export const receiveAlbums = (albums) => ({
  type: RECEIVE_ALBUMS,
  albums
});

export const requestAlbumsFailed = (error) => ({
  type: REQUEST_ALBUMS_FAILED,
  error
});

export const fetchAlbums = () => {
  return dispatch => {
    dispatch(requestAlbums());
    return fetch(FETCH_URL + "/albums")
      .then(res => res.json())
      .then(
        (result) => {
          const result1 = Object.values(result);
          /* eslint-disable array-callback-return */
          let result2 = [];
          result1.map((item, index) => {
            if (index === 0) {
              result2.push({text: item.name, path: "/"});
            } else {
              result2.push({text: item.name, path: "/" + item.id});
            }
          });
          dispatch(receiveAlbums(result2));
        },
        (error) => {
          dispatch(requestAlbumsFailed(error));
        }
      );
  }
}

export const requestData = () => ({
  type: REQUEST_DATA
});

export const receiveData = (data) => ({
  type: RECEIVE_DATA,
  data
});

export const requestDataFailed = (error) => ({
  type: REQUEST_DATA_FAILED,
  error
});

export const fetchData = (album) => {
  return dispatch => {
    dispatch(requestData());
    return fetch(FETCH_URL + "/albums/" + album)
      .then(res => res.json())
      .then(
        (result) => {
          dispatch(receiveData(result));
        },
        (error) => {
          dispatch(requestDataFailed(error));
        }
      );
  }
}

export const pageToActive = (page) => ({
  type: PAGE_TO_ACTIVE,
  activePage: page
});

export const toggleTheme = (theme) => ({
  type: TOGGLE_THEME,
  theme: theme
});

export const resize = (minHeight, deviceType) => ({
  type: RESIZE,
  minHeight: minHeight,
  deviceType: deviceType
});

