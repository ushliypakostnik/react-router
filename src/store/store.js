import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { browserHistory } from "react-router";
//import { createBrowserHistory } from "history";
import { syncHistoryWithStore, routerReducer } from 'react-router-redux';

import { INITIAL_STATE } from './constants';
import { pageToActive } from './actions';
import reducer from './reducers';

const middlewares = [];
middlewares.push(thunkMiddleware)

if (process.env.NODE_ENV !== 'production') {
  const loggerMiddleware = createLogger();

  middlewares.push(loggerMiddleware);
}

function configureStore(state) {
  return createStore(
    combineReducers({
      reducer,
      routing: routerReducer
    }),
    state,
    applyMiddleware(...middlewares)
  );
}

const store = configureStore(INITIAL_STATE);

//const history = createBrowserHistory();
export const history = syncHistoryWithStore(browserHistory, store);
//history.listen(location => console.info('-> location:', location));

let lastUrl = "";
if (store.getState().routing.locationBeforeTransitions != null) {
  lastUrl = store.getState().routing.locationBeforeTransitions.pathname;
} else {
  lastUrl = "/";
}

//store.subscribe(() => {
//  console.log("Store: ", store.getState());
//  console.log("last Url: ", lastUrl);
//});

store.dispatch(pageToActive(lastUrl));

export default store;
