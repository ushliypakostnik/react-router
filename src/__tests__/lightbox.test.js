import React from 'react';

import { FETCH_URL } from '../store/constants';

import LightboxContainer from '../components/lightbox';
import Lightbox from 'react-image-lightbox';

describe('lightbox render', () => {

  let wrapper,
      lightbox;

  const props = {
    images: [`${FETCH_URL}/media/pinhole/010.jpg`,
             `${FETCH_URL}/media/pinhole/020.jpg`,
             `${FETCH_URL}/media/pinhole/030.jpg`],
    key: 0,
    index: null,
    isOpen: false,
    lightboxUpdate: jest.fn()
  }

  const props2 = {
    ...props.images,
    mainSrc: props.images[1],
    nextSrc: props.images[0],
    prevSrc: props.images[2],
    onCloseRequest: jest.fn(),
    onMovePrevRequest: jest.fn(),
    onMoveNextRequest: jest.fn(),
    animationDuration: 100,
    enableZoom: false,
    reactModalStyl: {
      overlay: {
        zIndex: 2000
      }
    }
  }

  beforeEach(() => {
    wrapper = mount(<LightboxContainer {...props} />);
    lightbox = shallow(<Lightbox {...props2} />);
  });

  it('lightbox container render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('lightbox render correctly', () => {
    expect(toJson(lightbox)).toMatchSnapshot();
  });
});
