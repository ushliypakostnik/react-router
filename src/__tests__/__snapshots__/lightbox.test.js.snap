// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lightbox render lightbox container render correctly 1`] = `
<LightboxContainer
  images={
    Array [
      "http://127.0.0.1:8082/media/pinhole/010.jpg",
      "http://127.0.0.1:8082/media/pinhole/020.jpg",
      "http://127.0.0.1:8082/media/pinhole/030.jpg",
    ]
  }
  index={null}
  isOpen={false}
  lightboxUpdate={[MockFunction]}
/>
`;

exports[`lightbox render lightbox render correctly 1`] = `
<Modal
  appElement={<body />}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentLabel="Lightbox"
  isOpen={true}
  onAfterOpen={[Function]}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "backgroundColor": "transparent",
        "border": "none",
        "borderRadius": 0,
        "bottom": 0,
        "left": 0,
        "overflow": "hidden",
        "padding": 0,
        "right": 0,
        "top": 0,
      },
      "overlay": Object {
        "backgroundColor": "transparent",
        "zIndex": 1000,
      },
    }
  }
>
  <div
    className="ril-outer ril__outer ril__outerAnimating  "
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    onWheel={[Function]}
    style={
      Object {
        "animationDirection": "reverse",
        "animationDuration": "100ms",
        "transition": "opacity 100ms",
      }
    }
    tabIndex="-1"
  >
    <div
      className="ril-inner ril__inner"
      onClick={[Function]}
    >
      <div
        className="ril-image-next ril__imageNext ril__image ril-not-loaded"
        key="http://127.0.0.1:8082/media/pinhole/010.jpgi1"
        style={
          Object {
            "transform": "translate3d(1024px,0px,0) scale3d(NaN,NaN,1)",
          }
        }
      >
        <div
          className="ril__loadingContainer"
        >
          <div
            className="ril-loading-circle ril__loadingCircle ril__loadingContainer__icon"
          >
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="0"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="1"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="2"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="3"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="4"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="5"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="6"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="7"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="8"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="9"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="10"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="11"
            />
          </div>
        </div>
      </div>
      <div
        className="ril-image-current ril__image ril-not-loaded"
        key="http://127.0.0.1:8082/media/pinhole/020.jpgi0"
        style={
          Object {
            "transform": "translate3d(0px,0px,0) scale3d(NaN,NaN,1)",
          }
        }
      >
        <div
          className="ril__loadingContainer"
        >
          <div
            className="ril-loading-circle ril__loadingCircle ril__loadingContainer__icon"
          >
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="0"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="1"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="2"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="3"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="4"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="5"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="6"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="7"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="8"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="9"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="10"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="11"
            />
          </div>
        </div>
      </div>
      <div
        className="ril-image-prev ril__imagePrev ril__image ril-not-loaded"
        key="http://127.0.0.1:8082/media/pinhole/030.jpgi-1"
        style={
          Object {
            "transform": "translate3d(-1024px,0px,0) scale3d(NaN,NaN,1)",
          }
        }
      >
        <div
          className="ril__loadingContainer"
        >
          <div
            className="ril-loading-circle ril__loadingCircle ril__loadingContainer__icon"
          >
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="0"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="1"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="2"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="3"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="4"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="5"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="6"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="7"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="8"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="9"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="10"
            />
            <div
              className="ril-loading-circle-point ril__loadingCirclePoint"
              key="11"
            />
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Previous image"
      className="ril-prev-button ril__navButtons ril__navButtonPrev"
      key="prev"
      onClick={[Function]}
      type="button"
    />
    <button
      aria-label="Next image"
      className="ril-next-button ril__navButtons ril__navButtonNext"
      key="next"
      onClick={[Function]}
      type="button"
    />
    <div
      className="ril-toolbar ril__toolbar"
    >
      <ul
        className="ril-toolbar-left ril__toolbarSide ril__toolbarLeftSide"
      >
        <li
          className="ril-toolbar__item ril__toolbarItem"
        >
          <span
            className="ril-toolbar__item__child ril__toolbarItemChild"
          />
        </li>
      </ul>
      <ul
        className="ril-toolbar-right ril__toolbarSide ril__toolbarRightSide"
      >
        <li
          className="ril-toolbar__item ril__toolbarItem"
        >
          <button
            aria-label="Close lightbox"
            className="ril-close ril-toolbar__item__child ril__toolbarItemChild ril__builtinButton ril__closeButton"
            key="close"
            onClick={[Function]}
            type="button"
          />
        </li>
      </ul>
    </div>
  </div>
</Modal>
`;
